modules/cli/cli.go

```diff
@@ -62,9 +62,12 @@ func New(m metadata.Info) *CLI {
 	flagSet.UintVar(&cfg.KonnectMaxConcurrentReconciles, "konnect-controller-max-concurrent-reconciles", consts.DefaultKonnectMaxConcurrentReconciles, "Maximum number of concurrent reconciles for Konnect entities.")
 
 	// webhook and validation options
-	flagSet.BoolVar(&cfg.ValidatingWebhookEnabled, "enable-validating-webhook", false, "Enable the validating webhook.")
-	flagSet.StringVar(&cfg.WebhookCertificateConfigBaseImage, "webhook-certificate-config-base-image", consts.WebhookCertificateConfigBaseImage, "The base image for the certgen Jobs.")
-	flagSet.StringVar(&cfg.WebhookCertificateConfigShellImage, "webhook-certificate-config-shell-image", consts.WebhookCertificateConfigShellImage, "The shell image for the certgen Jobs.")
+	var validatingWebhookEnabled bool
+	flagSet.BoolVar(&validatingWebhookEnabled, "enable-validating-webhook", false, "Enable the validating webhook. DEPRECATED: This flag is no-op and will be removed in a future release.")
+	var validatingWebhookConfigBaseImage string
+	flagSet.StringVar(&validatingWebhookConfigBaseImage, "webhook-certificate-config-base-image", consts.WebhookCertificateConfigBaseImage, "The base image for the certgen Jobs. DEPRECATED: This flag is no-op and will be removed in a future release.")
+	var validatingWebhookConfigShellImage string
+	flagSet.StringVar(&validatingWebhookConfigShellImage, "webhook-certificate-config-shell-image", consts.WebhookCertificateConfigShellImage, "The shell image for the certgen Jobs. DEPRECATED: This flag is no-op and will be removed in a future release.")
```

Comment:
There is `flagSet.MarkDeprecated(...)` that has nice features like:
- do not print a marked flag in help
- prints a warning about deprecation when a flag is used (but doesn't stop program)
we've already used it in KIC

Reply:
This is only available in `github.com/spf13/pflag` not in https://pkg.go.dev/flag#FlagSet, sadly.

We'd have to change the underlying implementation. I'm open to doing this but ideally that'd be a PR on its own. WDYT?

`github.com/spf13/pflag` describes itself as being a drop in replacement so I believe that'd be the experience.